tests/monitoring_test.go

```diff
@@ -0,0 +1,135 @@
+package tests_test
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+const (
+	dataVolumeName = "test-dv"
+	validURL       = "http://www.example.com/example.img"
+)
+
+var _ = Describe("Monitoring Tests", func() {
+	f := framework.NewFramework("monitoring-test")
+
+	BeforeEach(func() {
+		if !f.IsPrometheusAvailable() {
+			Skip("This test depends on prometheus infra being available")
+		}
+	})
+
+	Context("Alerts", func() {
+		var crModified bool
+		var cr *cdiv1.CDI
+		var cdiPods *corev1.PodList
+
+		BeforeEach(func() {
+			cr = getCDI(f)
+			cdiPods = getCDIPods(f)
+		})
+
+		AfterEach(func() {
+			if crModified {
+				removeCDI(f, cr)
+				ensureCDI(f, cr, cdiPods)
+				crModified = false
+			}
+		})
+
+		It("should throw CDIDataVolumeUnusualRestartCount alert when a DataVolume object restarts more than three times", func() {
+			By("creating a DataVolume")
+			dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, "500Mi", validURL)
+			dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+			Expect(err).ToNot(HaveOccurred())
+			f.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
+
+			By("waiting for the DataVolume to be restarted more than three times")
+			Eventually(func() int32 {
+				dv, err := f.CdiClient.CdiV1beta1().DataVolumes(f.Namespace.Name).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return dv.Status.RestartCount
+			}).Should(BeNumerically(">", 3))
+
+			By("checking that the CDIDataVolumeUnusualRestartCount alert is firing")
+			checkAlertExists(f, "CDIDataVolumeUnusualRestartCount")
+
+			By("deleting the DataVolume")
+			err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should throw CDIStorageProfilesIncomplete alert if there is a incomplete StorageProfile", func() {
+			By("creating a StorageProfile with an incomplete configuration")
+			sp := &cdiv1.StorageProfile{
```

Comment:
I think we may be better using storage classes here to close out the flow (profile gets created for any class)

This is one of the reasons I am advocating using the existing tests.
We already have them cleaning up the remaining storageclass/profile after finishing
https://github.com/kubevirt/containerized-data-importer/blob/0de15cc7298921f96311d1bd2e8b079b14a06990/tests/operator_test.go#L807